pkg/app/pipedv1/plugin/scriptrun/plugin_test.go

```diff
@@ -256,14 +263,69 @@ func TestPlugin_ExecuteScriptRun(t *testing.T) {
 					ApplicationID: "app-3",
 				},
 			},
-			lp:   logpersistertest.NewTestLogPersister(t),
+			lp: logpersistertest.NewTestLogPersister(t),
+			metadataStore: mockDeploymentMetadataStore{
+				metadata: map[string]string{},
+			},
 			want: sdk.StageStatusFailure,
 		},
 	}
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
 			t.Parallel()
-			resp := executeScriptRun(t.Context(), tc.req, tc.lp)
+			resp := executeScriptRun(t.Context(), tc.req, tc.lp, &tc.metadataStore)
+			assert.Equal(t, tc.want, resp)
+		})
+	}
+}
+func TestPlugin_ExecuteRollback(t *testing.T) {
+	t.Parallel()
+	testcases := []struct {
+		name          string
+		req           sdk.ExecuteStageRequest[struct{}]
+		lp            sdk.StageLogPersister
+		metadataStore mockDeploymentMetadataStore
+		want          sdk.StageStatus
+	}{
+		{
+			name: "rollback run if its scriptrun record exists",
+			req: sdk.ExecuteStageRequest[struct{}]{
+				StageName:   stageScriptRunRollback,
+				StageIndex:  1,
+				StageConfig: []byte(`{"run": "echo 4", "onRollback": "echo 'rollback'"}`),
+				Deployment: sdk.Deployment{
+					ID:            "deployment-4",
+					ApplicationID: "app-4",
+				},
+			},
+			lp: logpersistertest.NewTestLogPersister(t),
+			metadataStore: mockDeploymentMetadataStore{
+				metadata: map[string]string{"1": "_"},
+			},
+			want: sdk.StageStatusSuccess,
+		},
+		{
+			name: "rollback should not run if its scriptrun record does not exists",
+			req: sdk.ExecuteStageRequest[struct{}]{
+				StageName:   stageScriptRunRollback,
+				StageIndex:  1,
+				StageConfig: []byte(`{"run": "echo 5","onRollback": "this would fail if run"}`),
```

Comment:
This case seems not enough for testing the behavior of checking the metadata since it will success even if the metadata does not exist.

what about this one?: 
```suggestion
				StageConfig: []byte(`{"run": "echo 5","onRollback": "exit 1"}`),
```

Reply:
i think both serve the same purpose (the rollback script won't trigger if no metadata found)

Reply:
@hiep-tk no. if the metadata logic is failed, `exit 1` will be executed.